rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol

```diff
@@ -456,45 +446,57 @@ contract RBTCWrapperProxy is ContractRegistryClient {
-        liquidityMiningContract.deposit(loanTokenAddress, minted, msg.sender);   
-
-        emit LoanTokensMinted(msg.sender, minted, depositAmount);   
+    function _refund(
+        address _liquidityPoolConverterAddress,
+        IERC20Token[] memory _reserveTokens,
+        uint256[] memory _initialProxyBalances
+    ) private {
+        for (uint256 i = 0; i < _reserveTokens.length; i++) {            
+            uint256 latestTokenBalance = _reserveTokens[i].balanceOf(address(this));
+            uint256 amount = latestTokenBalance.sub(_initialProxyBalances[i]);
+            if (amount > 0) {
+                require(_reserveTokens[i].approve(_liquidityPoolConverterAddress, 0), "Failed to reset approval");
+                if(address(_reserveTokens[i]) == wrbtcTokenAddress ) {
+                    IWrbtcERC20(wrbtcTokenAddress).withdraw(amount);
+                    (bool success,) = msg.sender.call.value(amount)("");
+                    require(success, "Failed to send extra RBTC back to user");
+                } else {
+                    require(_reserveTokens[i].transfer(msg.sender, amount), "Failed to transfer extra reserve token back to user");
+                }
+            }
+        }
     }
 
     /**
-     * @notice removes funds from the liquidity mining contract, burns them on the lending pool and 
-     * provides the underlying asset to the user
-     * @param loanTokenAddress the address of the loan token (aka lending pool)
-     * @param burnAmount the amount of pool tokens to withdraw from the lending pool and burn
+     * @dev Owner withdraw stuck token / rbtc in this contract.
+     *
+     * @param token The token address that will be withdrawn. Pass 0 address for withdrawing rbtc.
+     * @param to The recipient of withdrawal.
+     * @param amount amount to the withdraw token / rbtc
      */
-    function removeFromLendingPool(address loanTokenAddress, uint256 burnAmount) public{
-        LoanToken loanToken = LoanToken(loanTokenAddress);
-
-        //withdraw always transfers the pool tokens to the caller and the reward tokens to the passed address
-        liquidityMiningContract.withdraw(loanTokenAddress, burnAmount, msg.sender);
-
-        //burn pool token and directly send underlying tokens to the receiver
-        loanToken.approve(address(liquidityMiningContract), burnAmount);
-        uint256 redeemed = loanToken.burn(msg.sender, burnAmount);
+    function withdraw(address token, address payable to, uint256 amount) external ownerOnly {
+        require(amount > 0, "non-zero withdraw amount expected");
+        require(to != address(0), "receiver address invalid");
+
+        if (token == address(0)) {
+            require(amount <= address(this).balance, "withdraw amount cannot exceed balance");
+            to.transfer(amount);
+        } else {
+            require(IERC20Token(token).transfer(to, amount), "Failed to transfer the token");
+        }
```

Comment:
CD6 - Don't use assert, tx.origin, address.transfer(), address.send()
https://consensys.net/diligence/blog/2019/09/stop-using-soliditys-transfer-now/
```
import "../openzeppelin/SafeERC20.sol"; // copy from the protocol or from OZ - install proper version
```
```
using SafeERC20 for IERC20; // in the contract
```
address to;  - to allow transfer to a contract address too (e.g. multisig)
```suggestion
   function withdraw(
        address token,
        address to, 
        uint256 amount
    ) external ownerOnly {
        require(amount > 0, "non-zero withdraw amount expected");
        require(to != address(0), "receiver address invalid");
        if (token == address(0)) {
            (bool success, ) = to.call.value(amount)("");
            require(success, "transfer failed");
        } else {
            IERC20(token).safeTransfer(token, to, amount);
        }
    }
```